images/router/haproxy/shutdown-haproxy

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+
+set -euo pipefail
+
+# TERMINATION_MARKER is the path to a file that indicates that this router is
+# shutting down.  It is used as follows:
+#
+# - The reload-haproxy script exits immediately if the marker is present.
+#
+# - This script performs a hard shutdown if the marker is already present, or
+#   else it sets the marker and performs a graceful shutdown.
+TERMINATION_MARKER='/var/lib/haproxy/run/terminating'
+
+# DRAIN_PERIOD is the number of seconds that this script will wait for
+# connections to drain before it sends the SIGUSR1 signal to HAProxy processes.
+DRAIN_PERIOD=30
+
+# GRACE_PERIOD is the number of seconds that this script will wait for HAProxy
+# processes to terminate after it sends the SIGUSR1 signal, before it sends
+# SIGTERM to any remaining HAProxy processes.
+GRACE_PERIOD=30
+
+graceful_shutdown='false'
+if ! [[ -e "$TERMINATION_MARKER" ]]; then
+    : > "$TERMINATION_MARKER"
+    graceful_shutdown='true'
+fi
+
+if [[ "$graceful_shutdown" = 'true' ]]; then
+    echo 'Performing a graceful shutdown:'
+    echo " - Sleeping $DRAIN_PERIOD seconds to let connections drain..."
+    sleep "$DRAIN_PERIOD"
```

Comment:
Granted the reload script will be no-op'ing, if we haven't issued USR1 to haproxy it's not clear to me how connection draining is implemented with this sleep.